pkg/kvstore/etcd.go

```diff
@@ -101,12 +138,33 @@ func NewEtcdClient(config *client.Config, cfgPath string) (KVClient, error) {
 				ec.session = newSession
 				ec.sessionMU.Unlock()
 				log.Debugf("Renewing etcd session")
+				ec.CheckMinVersion()
 			}
 		}
 	}()
 	return ec, nil
 }
 
+// CheckMinVersion checks the minimal version running on etcd cluster. If the
+// minimal version running doesn't meet cilium minimal requirements, exits
+// cilium.
+func (e *EtcdClient) CheckMinVersion() {
+	eps := e.cli.Endpoints()
+	success := false
+	for _, ep := range eps {
+		if err := checkMinVersionOfEndpoint(ep); err != nil {
+			log.Debugf("Unable to check min version: %s", err)
+		} else {
+			success = true
+			break
+		}
+	}
+	if !success {
+		log.Warningf("Unable to check etcd's cluster version."+
```

Comment:
That's just a `Warning`. We check all endpoint for the cluster version. The cluster version is the minimal etcd version running on the cluster. We just need to check the cluster version from a single endpoint, if we can't access any endpoint then we print a warning.

I understand that we should not bail out if the minimal version doesn't meet the requirements while cilium is running. The problem is that we need to make enhancements in the kvstore lib to re-fetch all security IDs from the kvstore, otherwise we might end with 2 different IDs for the same set of labels in the cluster.

For example, if kv-store connection is lost, the container's keep-alive that we set in the `SecLabelID` is no longer refreshed, setting that label as unused.